scheduler/src/cook/scheduler/scheduler.clj

```diff
@@ -1084,19 +1084,64 @@
                               :first-ten-jobs (print-str (->> jobs (take 10) (map :job/uuid) (map str)))}))
       (dissoc compute-cluster->jobs :no-acceptable-compute-cluster))))
 
+(defn decrement-scheduling-capacity-counter
+  "Decrement capacity for compute cluster in the counter."
+  [capacity-counter compute-cluster]
+  (update-in capacity-counter [compute-cluster] dec))
+
+(defn scheduling-capacity-constrained-job-distributor
+  "Create a job distributor function that only assigns jobs to compute clusters
+   with available scheduling capacity. Assignments are tracked and the given
+   counter is decremented."
+  [compute-cluster->scheduling-capacity autoscalable-jobs pool-name compute-clusters job->acceptable-compute-clusters-fn]
+  (tracing/with-span [s {:name "scheduler.distribute-jobs-to-compute-cluster"
+                         :tags {:pool pool-name :component tracing-component-tag}}]
+    (let [capacity-counter-atom (atom compute-cluster->scheduling-capacity)
+          compute-cluster->jobs
+          (group-by
+           (fn choose-compute-cluster-for-autoscaling
+             [{:keys [job/uuid] :as job}]
```

Comment:
Job UUID or instance UUID? I'd vote for instance UUID because it should make reruns route to a new CC, for hopefully better luck there.

Reply:
At this point, we do not have an instance for the job. Only after this function assigns do we create that instance and then launch. But that's a really interesting point. Do you know why we shouldn't just make this random?

Reply:
I see no reason not to make it random as long as this code isn't called from the synthetic-pod codepath. (And AFACT, its not. Can you confirm?), so excellent suggestion to make this random.

Reply:
Yup this code is just for new Kenzo side. Done. (rand-nth is nice)